### PR TITLE
Add default routes over TGW

### PR DESCRIPTION
**Pull request by snovikov**
# Add default routes over TGW

## Description
 Make possible use TGW as a default route for private subnets

## Notes
⚠️ Backwards compatibility will be broken!
This is up for a major release.

### CODE REVIEW

File: examples/satellite/variables.tf [hunk 1]
```diff
@@ -46,8 +46,28 @@ variable "hub_destination_cidr_blocks" {
...
+variable "attachment_subnet_filters" {
+  description = "List of maps selecting the subnet(s) where TGW will be attached"
+  type = list(object({
+    name   = string
+    values = list(string)
+  }))
+  default = [
+    {
+      name   = "tag:Name"
+      values = ["private"]
```
Comment (danvaida) [inline examples/satellite/variables.tf]:
```suggestion
      values = ["*private*"]
```

File: examples/satellite/variables.tf [hunk 2]
```diff
@@ -46,8 +46,28 @@ variable "hub_destination_cidr_blocks" {
...
+    }
+  ]
+}
+
+variable route_private_subnets_via_tgw {
```
Comment (danvaida) [inline examples/satellite/variables.tf]:
```suggestion
variable "route_private_subnets_via_tgw" {
```

File: variables.tf [hunk 3]
```diff
@@ -66,8 +66,28 @@ variable "ram_resource_association_id" {
...
+variable "attachment_subnet_filters" {
+  description = "List of maps selecting the subnet(s) where TGW will be attached"
+  type = list(object({
+    name   = string
+    values = list(string)
+  }))
+  default = [
+    {
+      name   = "tag:Name"
+      values = ["private"]
```
Comment (danvaida) [inline variables.tf]:
```suggestion
      values = ["*private*"]
```

File: examples/satellite/main.tf [hunk 4]
```diff
@@ -22,9 +22,13 @@ module "tgw-satellite" {
...
   hub_destination_cidr_blocks       = var.hub_destination_cidr_blocks
 
-  subnet_filters = var.subnet_filters
+  attachment_subnet_filters = var.attachment_subnet_filters
 
   transit_gateway_hub_name = var.transit_gateway_hub_name
 
   route_entire_satellite_vpc = var.route_entire_satellite_vpc
+
+  route_private_subnets_via_tgw = var.route_private_subnets_via_tgw
```
Comment (danvaida) [inline examples/satellite/main.tf:31]: Would be great to have a separate test case for this new functionality, while keeping the existing test case in place, to ensure that the old behaviour is still working as expected.

**Comment by snovikov** [comment]
@danvaida I also found an issue if 
`route_entire_satellite_vpc = false`

```
   8:   routing_tables_all     = data.aws_route_tables.all[0].ids
    |----------------
    | data.aws_route_tables.all is empty tuple

The given key does not identify an element in this collection value.
```

File: examples/satellite/main.tf (hunk 4 above)
Comment (snovikov) [reply examples/satellite/main.tf:31]: @danvaida I added a separate test case

File: examples/satellite/variables.tf (hunk 1 above)
Comment (snovikov) [reply examples/satellite/variables.tf]: done

File: examples/satellite/variables.tf (hunk 2 above)
Comment (snovikov) [reply examples/satellite/variables.tf]: done

File: variables.tf (hunk 3 above)
Comment (snovikov) [reply variables.tf]: done

**Review by danvaida — APPROVED** [review]

**Pull request merged**